VocableUITests/Tests/SettingsScreenTests.swift

```diff
@@ -3,45 +3,42 @@
-
-        settingsScreen.navigateToCategory(category: hiddenGeneralCategoryText)
-
-        XCTAssertFalse(settingsScreen.otherElements.containing(.staticText, identifier: hiddenGeneralCategoryText).buttons[settingsScreen.settingsPageCategoryUpButton].isEnabled)
-        XCTAssertFalse(settingsScreen.otherElements.containing(.staticText, identifier: hiddenGeneralCategoryText).buttons[settingsScreen.settingsPageCategoryDownButton].isEnabled)
-        XCTAssert(settingsScreen.otherElements.containing(.staticText, identifier: hiddenGeneralCategoryText).buttons[settingsScreen.settingsPageCategoryShowButton].isEnabled)
-
-        // Verify category goes back to original spot when shown.
-
-        settingsScreen.toggleHideShowCategory(category: hiddenGeneralCategoryText, toggle: "Show")
-        XCTAssertFalse(settingsScreen.otherElements.containing(.staticText, identifier: hiddenGeneralCategoryText).element.exists)
-
-        settingsScreen.navigateToCategory(category: generalCategoryText)
-        settingsScreen.settingsPageNextButton.tap()
-
-        XCTAssertFalse(settingsScreen.otherElements.containing(.staticText, identifier: generalCategoryText).buttons[settingsScreen.settingsPageCategoryUpButton].isEnabled)
-        XCTAssert(settingsScreen.otherElements.containing(.staticText, identifier: generalCategoryText).buttons[settingsScreen.settingsPageCategoryDownButton].isEnabled)
-        XCTAssert(settingsScreen.otherElements.containing(.staticText, identifier: generalCategoryText).buttons[settingsScreen.settingsPageCategoryHideButton].isEnabled)
+        // Verify that when the category is hidden, up and down buttons are disabled.
+        settingsScreen.openCategorySettings(category: category)
+        settingsScreen.showCategorySwitch.tap()
+        settingsScreen.leaveCategoryDetailButton.tap()
+        
+        settingsScreen.locateCategoryCell(category)
+        XCTAssertFalse(settingsScreen.locateCategoryCell(category).buttons[settingsScreen.categoryUpButton].isEnabled)
+        XCTAssertFalse(settingsScreen.locateCategoryCell(category).buttons[settingsScreen.categoryDownButton].isEnabled)
+        XCTAssertTrue(settingsScreen.locateCategoryCell(category).buttons[settingsScreen.categoryForwardButton].isEnabled)
+
+        // Verify that when the category is shown, up and down buttons are enabled.
+        settingsScreen.locateCategoryCell(category).buttons[settingsScreen.categoryForwardButton].tap()
+        settingsScreen.showCategorySwitch.tap()
+        settingsScreen.leaveCategoryDetailButton.tap()
+        
+        settingsScreen.locateCategoryCell(category)
+        XCTAssertTrue(settingsScreen.locateCategoryCell(category).buttons[settingsScreen.categoryUpButton].isEnabled)
+        XCTAssertTrue(settingsScreen.locateCategoryCell(category).buttons[settingsScreen.categoryDownButton].isEnabled)
+        XCTAssertTrue(settingsScreen.locateCategoryCell(category).buttons[settingsScreen.categoryForwardButton].isEnabled)
```

Comment:
```suggestion
        let categoryCell = settingsScreen.locateCategoryCell(category)
        guard categoryCell.waitForExistence(20) else 
            XCTFail("Couldn't locate category cell")
            return
        }        
        XCTAssertTrue(categoryCell.buttons[settingsScreen.categoryUpButton].isEnabled)
        XCTAssertTrue(categoryCell.buttons[settingsScreen.categoryDownButton].isEnabled)
        XCTAssertTrue(categoryCell.buttons[settingsScreen.categoryForwardButton].isEnabled)
```
Just using this one snippet as an example, I think you can grab the cell element once and store it as a variable to reference in the assertions below it. I also threw in a `waitForExistence` because I'm not sure exactly what happens if `locateCategoryCell()` is unsuccessful, but [there are other strategies](https://mokacoding.com/blog/write-better-swift-tests-with-xctest-assertions/) that might help with that. Not a huge deal, mostly just wanted to grab the cell once for performance and readability reasons 😄

Reply:
We try to “handle" what happens if the cell is not present initially in the loop within `locateCategoryCell()`. We don't throw a failure explicitly but we do let the query fail which will report that the cell could not be found. Before that, however, we scroll the pages within the list of categories to find the cell we're looking for. 

Making a reference to the cell once in order to shorten our run time and the line of code is a good idea. We will implement that.

Reply:
Fixed.